ScopeLift/umbra-protocol
# Update private key management

Closes #44 and supports #29 

The `refactor-frontend` branch is a long running branch to rebuild the frontend with the Composition API and Typescript, and will continually be rebased against master. The `app` folder contains the frontend code that exists on master and will ultimately be deleted. The `frontend` folder contains the new frontend code that is being built on the `refactor-frontend` branch. The frontend in the new `frontend` folder is not yet complete and functionality will be added to it as needed.

In this PR we make the following changes to the `frontend` folder
- Add send and receive pages
- Setup linting/formatting on commit (also makes sure app builds on each commit)
- Remove Login button from header. Instead the user is now prompted to connect wallet and sign any required messages when choosing either "Send" or "Receive" from the home page
- Add route guards so you cannot navigate directly to the send or receive pages. You must always start at the home page. This is the simplest solution for now to ensure a user always connects a wallet and signs any required messages
- Visiting the Send page uses the logic/store in `keys.ts` to prompt the user for their signature and save their private keys in memory

To test:
- Checkout this branch
- Run `yarn` to install packages
- Run `yarn dev` to start the app
- Visit the home page, click "Send", sign the message, and make sure both key pairs are logged to the console
